templates/secrets_st2auth.yaml

```diff
@@ -15,6 +22,10 @@ metadata:
 type: Opaque
 data:
   # Username, used to login to StackStorm system (default: st2admin)
-  username: {{ required "A valid secret 'st2.username' is required for StackStorm auth!" .Values.secrets.st2.username | b64enc | quote }}
-  # Password, used to login to StackStorm system (default: Ch@ngeMe)
-  password: {{ required "A valid secret 'st2.password' is required for StackStorm auth!" .Values.secrets.st2.password | b64enc | quote }}
+  username: {{ required "A valid secret 'st2.username' is required for StackStorm auth!" .Values.st2.username | b64enc | quote }}
+  # Password, used to login to StackStorm system (default: auto-generated)
+{{- if .Release.IsUpgrade }}
+  password: {{ index (lookup "v1" "Secret" .Release.Namespace $name).data "password" }}
+{{ else }}
+  password: {{ default (randAlphaNum 12) .Values.st2.password | b64enc | quote }}
+{{ end }}
```

Comment:
I hadn't chance to explore deeper the new `lookup` functionality.

How this thing will behave if user needs to override the previous password set?

Any edge cases?

Reply:
Hmm. We might need a flag like `resetPassword` which would make it skip the lookup. Something like this:
```
{{- if and .Release.IsUpgrade (not .Values.st2.resetPassword) }}
```

Wdyt?

Reply:
One edge case might be trying to update the password while upgrading from an existing install without setting that flag. I can't think of anything other edge cases. You?

Reply:
fixed

Reply:
Ah, replied in another thread: https://github.com/StackStorm/stackstorm-ha/pull/203/files#r660046629